src/main/java/appeng/items/storage/ItemStorageCellBase.java

```diff
@@ -0,0 +1,271 @@
+ * @since rv6 2018-01-17
+ */
+public abstract class ItemStorageCellBase<T extends IAEStack<T>> extends AEBaseItem implements IStorageCell<T>, IItemGroup
+{
+	protected final MaterialType component;
+	protected final int totalBytes;
+
+	public ItemStorageCellBase( final MaterialType whichCell, final int kilobytes )
+	{
+		this.setMaxStackSize( 1 );
+		this.totalBytes = kilobytes * 1024;
+		this.component = whichCell;
+	}
+
+	@SideOnly( Side.CLIENT )
+	@Override
+	public void addCheckedInformation( final ItemStack stack, final World world, final List<String> lines, final ITooltipFlag advancedTooltips )
+	{
+		final IMEInventoryHandler<?> inventory = AEApi.instance().registries().cell().getCellInventory( stack, null, getChannel() );
+
+		if( inventory instanceof ICellInventoryHandler )
+		{
+			final ICellInventoryHandler handler = (ICellInventoryHandler) inventory;
+			final ICellInventory cellInventory = handler.getCellInv();
+
+			if( cellInventory != null )
+			{
+				lines.add( cellInventory.getUsedBytes() + " " + GuiText.Of.getLocal() + ' ' + cellInventory.getTotalBytes() + ' ' + GuiText.BytesUsed.getLocal() );
+
+				lines.add( cellInventory.getStoredItemTypes() + " " + GuiText.Of.getLocal() + ' ' + cellInventory.getTotalItemTypes() + ' ' + GuiText.Types.getLocal() );
+
+				if( handler.isPreformatted() )
+				{
+					final String list = ( handler.getIncludeExcludeMode() == IncludeExclude.WHITELIST ? GuiText.Included : GuiText.Excluded ).getLocal();
+
+					if( handler.isFuzzy() )
+					{
+						lines.add( GuiText.Partitioned.getLocal() + " - " + list + ' ' + GuiText.Fuzzy.getLocal() );
+					}
+					else
+					{
+						lines.add( GuiText.Partitioned.getLocal() + " - " + list + ' ' + GuiText.Precise.getLocal() );
+					}
+				}
+			}
+		}
+	}
+
+	@Override
+	public int getBytes( final ItemStack cellItem )
+	{
+		return this.totalBytes;
+	}
+
+	@Override
+	public int getTotalTypes( final ItemStack cellItem )
+	{
+		return 63;
+	}
+
+	@Override
+	public boolean isBlackListed( final ItemStack cellItem, final T requestedAddition )
+	{
+		return false;
+	}
+
+	@Override
+	public boolean storableInStorageCell()
+	{
+		return false;
+	}
+
+	@Override
+	public boolean isStorageCell( final ItemStack i )
+	{
+		return true;
+	}
+
+	@Override
+	public String getUnlocalizedGroupName( final Set<ItemStack> others, final ItemStack is )
+	{
+		return GuiText.StorageCells.getUnlocalized();
+	}
+
+	@Override
+	public boolean isEditable( final ItemStack is )
+	{
+		return true;
+	}
+
+	@Override
+	public IItemHandler getUpgradesInventory( final ItemStack is )
+	{
+		return new CellUpgrades( is, 2 );
+	}
+
+	@Override
+	public IItemHandler getConfigInventory( final ItemStack is )
+	{
+		return new CellConfig( is );
+	}
+
+	@Override
+	public FuzzyMode getFuzzyMode( final ItemStack is )
+	{
+		final String fz = Platform.openNbtData( is ).getString( "FuzzyMode" );
+		try
+		{
+			return FuzzyMode.valueOf( fz );
+		}
+		catch( final Throwable t )
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Either log or rethrow this exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1166)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Catch Exception instead of Throwable. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1181)